050-Kafka-Node-Pools.md

```diff
@@ -0,0 +1,806 @@
+So when the use specifies with annotation for example that new nodes should come from a range `[1000-1010]`, such annotation is considered valid even when some of the IDs from the range are already in use.
+This way, the node pool can auto-scale up and down within the range without changing the annotation.
+
+#### Moving nodes between node pools
+
+In some situations, it might be good to be able to move a node between different node pools.
+The annotations can be used for that as well.
+The process to move node with ID 3 from pool named `small-nodes` to pool names `big-nodes` will be following:
+1) Annotate `KafkaNodePool` named `small-nodes` with annotation `strimzi.io/remove-node-ids: [3]`
+2) Annotate `KafkaNodePool` named `big-nodes` with annotation `strimzi.io/next-node-ids: [3]`
+3) Scale `KafkaNodePool` named `small-nodes` down by changing its `.spec.replicas` field
+4) Scale `KafkaNodePool` named `big-nodes` up by changing its `.spec.replicas` field
+
+The operator will proceed and remove the node 3 from the `small-nodes` pool and create a new node with ID 3 in the `big-nodes` pool.
+The moved now in its new pod will start from zero and will have to replicate all the data from other nodes in the cluster.
+
+This process has several risks:
+* The moved node starts from 0 with empty disk and will need to re-sync all data from other nodes.
+  Kafka nodes might store in some environments TBs of data, so re-syncing the data might take a long time.
+* While the node is re-syncing, the replicas hosted by it are not in-sync.
+  So with typical configurations (such as replication factor set to 3 and minimal in-sync replicas 2) any problem with one of the other node would mean loss of availability and cause that producers will not be able to produce messages anymore.
+* Since the replica will not be in-sync, it would also mean that it might not be possible to proceed with rolling updates and other operations.
+
+Due to these risks and limitations, users should instead of moving a node consider moving the partition replicas.
+They should:
+1) Scale up the target node pool to have a new node created
+2) Reassign the partition replicas from the old node to the new node
+3) Once the reassignment is complete, scale down the node pool with the old node
+
+While this will cause a change of the node ID, it will be a more graceful process.
+The reassignment process will first create a new replica and start copying the data.
+Only when the new replica on the new node is in-sync, the old replica will be deleted.
+Thanks to that, the availability is still guaranteed and other changes such as rolling updates are not affected.
+This process should be also recommended to the users in the documentation.
+
+In the future, we should try to improve the Cruise Control support to make this process easier (maybe with a new rebalance mode `move-broker` or something similar).
```

Comment:
While reading above I was wondering the same, so we should make clear that all the process is possible manually using the reassignment tool.